docs/proposals/organizations.md

```diff
@@ -0,0 +1,125 @@
+# Organizations
+
+## Problems
+ 1.  We need to be able to allocate quota to a group of projects.
+ 2.  We need to be able to limit a particular project's usage of shared quota (think htb rate/ceil for tc qdisc classes).
+ 3.  We need to be able to manage groups across related projects, but not across the entire cluster.
+ 4.  We need to be able to limit how many projects a given User/Organization can have.  (limits provisioning)
+
+We can do this with an Organization entity that can manage multiple projects.  An Organization can have multiple org-admins who are allowed to manage quota allocation to owned projects and OrgGroups (groups that scoped to projects owned by an Organization).  A project is owned by at most one Organization, an Organization can own muliple projects.
+
+### Open Questions
+ 1.  Should we have a two layer scoping `/api/v1/organization/<org name>/namespace/<namespace name>` or keep the current namespace structure?
+
+     Keeping the current namespace structure mitigates changes, but prevents two orgs from having the same namespace and makes storing additional org-scoped resources like quota and groups more difficult.
+     I'm in favor of keeping the current structure, but allowing deeper nesting of subresources and having named subresources.
```

Comment:
Adding a new layer is not a composition pattern - it requires every client to change.  We could in the future expose a new prefix in order to solve certain problems.   The hard problem is access control.  However, we say that to find the things that you have access to, you have to be a member of the project anyway.

I would prefer to treat organization as a thing that controls / reacts to projects, rather than something that is deeply embedded into the project pattern.